core/src/dird/consolidate.cc

```diff
@@ -251,21 +251,23 @@ bool DoConsolidate(JobControlRecord* jcr)
                job->resource_name_);
           continue;
         }
-        Jmsg(jcr, M_INFO, 0, _("before ConsolidateFull: jobids: %s\n"), jobids);
+        Jmsg(jcr, M_INFO, 0, _("before ConsolidateFull: jobids: %s\n"),
+             jobids_ctx.GetAsString().c_str());
 
-        p = strchr(jobids, ','); /* find oldest jobid and optionally skip it */
-        if (p) { *p = '\0'; }
+        std::string oldestjobid = jobids_ctx.front();
```

Comment:
```suggestion
        std::string& oldestjobid = jobids_ctx.front();
```
There is no need to copy here since we only use it while the reference is still ok.